x/programs/rust/wasmlanche_sdk/src/types.rs

```diff
@@ -98,12 +102,24 @@ impl Argument for Bytes32 {
     fn as_bytes(&self) -> Cow<'_, [u8]> {
         Cow::Borrowed(&self.0)
     }
+    fn from_bytes(bytes: &[u8]) -> Self
+    where
+        Self: Sized,
+    {
+        let bytes = bytes[..ADDRESS_LEN].try_into().unwrap();
+        Self::new(bytes)
+    }
```

Comment:
You can do this everywhere and connect the proper error types later. 

```suggestion
    fn from_bytes(bytes: &[u8]) -> Result<Self, Box<dyn std::error::Error>>
    where
        Self: Sized,
    {
        let bytes = bytes[..ADDRESS_LEN].try_into()?;
        Self::new(bytes)
    }
```